app/javascript/portal/portalHelpers.js

```diff
@@ -115,6 +115,32 @@ export const InitializationHelpers = {
     });
   },
 
+  openExternalLinksInNewTab: () => {
+    document.querySelectorAll('a').forEach(link => {
+      const { hostname } = new URL(link.href);
+      const { customDomain, hostURL } = window.portalConfig;
+
+      const href = link.getAttribute('href');
+      const isExternalLink =
+        hostname !== window.location.hostname ||
+        href.startsWith('http') ||
+        href.startsWith('https') ||
+        href.startsWith('//');
+
+      const isSameHost =
+        window.location.href.includes(customDomain) ||
+        window.location.href.includes(hostURL);
+
+      // Modify external links only on articles page
+      const isOnArticlePage =
+        isSameHost && window.location.pathname.includes('/articles/');
+
+      if (isExternalLink && isOnArticlePage) {
+        link.target = '_blank';
+      }
+    });
+  },
+
```

Comment:
@nithindavid Can we add `isOnArticlePage` check before `document.querySelectorAll`. This can avoid unnecessary iterations if the user is not on an article page. Thoughts?

For example:
```
    if (isOnArticlePage) {
      document.querySelectorAll('a').forEach(link => {
    
        if (isExternalLink) {
          //
        }
      });
    }
```

Reply:
Thanks for the suggestion. I have added this.